docs/develop/writingtests.rst

```diff
@@ -0,0 +1,73 @@
+.. _writingtests:
+
+Writing tests
+=============
+
+Tests in MicroPython are written in the path ``py/tests``:
+
+.. code-block:: console
+   
+   .
+    ├── basics
+    ├── cmdline
+    ├── cpydiff
+    ├── esp32
+    ├── extmod
+    ├── feature_check
+    ├── float
+    ├── import
+    ├── inlineasm
+    ├── internal_bench
+    ├── io
+    ├── jni
+    ├── micropython
+    ├── misc
+    ├── multi_bluetooth
+    ├── multi_net
+    ├── net_hosted
+    ├── net_inet
+    ├── perf_bench
+    ├── pyb
+    ├── pybnative
+    ├── qemu-arm
+    ├── README
+    ├── run-internalbench.py
+    ├── run-multitests.py
+    ├── run-natmodtests.py
+    ├── run-perfbench.py
+    ├── run-tests
+    ├── run-tests-exp.py
+    ├── run-tests-exp.sh
+    ├── stress
+    ├── thread
+    ├── unicode
+    ├── unix
+    └── wipy
+
+There are subfolders maintained to categorize most tests. Add a test by creating a new file in one of the
+existing folders or in a new folder.
```

Comment:
We can add more detail, but perhaps the two key things to add here are:
* This whole thing works by comparing the output of a program by running it against the unix port and CPython. So your "test" should use print statements to indicate test results.
* For tests that can't be compared to CPython (i.e. micropython-specific functionality), you can provide a .exp file.